### PR TITLE
PHP 8.4 support

### PR DESCRIPTION
**Pull request by realFlowControl**
### Description

This PR will bring support for PHP 8.4 🎉 

PROF-10137

### Reviewer checklist
- [ ] Test coverage seems ok.
- [x] Appropriate labels assigned.

### CODE REVIEW

File: tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+include __DIR__ . '/../dd_dumper.inc';
+\dd_dump_spans(true);
+
+?>
+--EXPECTF--
+spans(\DDTrace\SpanData) (8) {
+  intval (trace_closure_ge_php_84.php, 0, cli)
+    result => 0
+    _dd.p.tid => %s
+  {closure:%s.php:7\{closure} (trace_closure_ge_php_84.php, 1, cli)
```
Comment (realFlowControl) [inline tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt:74]: @bwoebi this does not seem right, there is a missing closing bracket
```diff
- {closure:%s.php:7\{closure}
+ {closure:%s.php:7\{closure\}}
```
should maybe look something like this?

Comment (bwoebi) [reply tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt:74]: Why write "{closure}" twice? I think it should look like `{closure:%s.php:7}`, which is the format PHP 8.4 also uses for closures now.

Comment (realFlowControl) [reply tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt:74]: Sorry, I mean that 
```
  {closure:%s.php:7\{closure} (trace_closure_ge_php_84.php, 1, cli)
```
is the version that makes the test succeed, so I am thinking we might need some changes in the extension.

Comment (bwoebi) [reply tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt:74]: Yes, we do :-)

**Comment by morrisonlevi** [comment]
I've tentatively added a commit which removes the `zend_execute_internal` bits on PHP 8.4. It's probably a slight abuse of `cfg(php_frameless)` but whatever, allows us to test it.

**Review by bwoebi — APPROVED** [review]
We have 14 CI failures left. Nothing critical, something we theoretically might release with.
Let's merge it and improve upon it later.

**Pull request merged**